### PR TITLE
update: feature engineering unit tests

### PR DESCRIPTION
**Pull request by alxlyj**
### Summary

- Update feature eng unit test.

### Test Plan

image

image

### CODE REVIEW

File: python/tests/src/robyn/modeling/test_feature_engineering.py
Comment (sumane81) [inline python/tests/src/robyn/modeling/test_feature_engineering.py:1]: 1. Code coverage is at 70% compared to 74% from earlier tests. 
2. Can you remove the src folder from the path?

File: python/tests/src/robyn/modeling/test_feature_engineering.py [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+from robyn.data.entities.hyperparameters import Hyperparameters, ChannelHyperparameters
+from robyn.data.entities.holidays_data import HolidaysData
+from robyn.data.entities.enums import AdstockType
+
+
+class TestFeatureEngineering(unittest.TestCase):
+
+    def setUp(self):
+        # Create main mocks
+        self.mmm_data = MagicMock(spec=MMMData)
```
Comment (sumane81) [inline python/tests/src/robyn/modeling/test_feature_engineering.py:16]: I preferred using simulated data than letting unit test mock create high level mock data.

File: python/tests/src/robyn/modeling/test_feature_engineering.py [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+            "yhat": pd.DataFrame(),
+            "plots": {},
+        }
+
+        # Execution
+        result = self.feature_engineering.perform_feature_engineering()
+
+        # Assertions
+        self.assertIsInstance(result, FeaturizedMMMData)
+        self.assertTrue(hasattr(result, "dt_mod"))
```
Comment (sumane81) [inline python/tests/src/robyn/modeling/test_feature_engineering.py:78]: Just having the attributes is good enough test?

File: python/spec/src/robyn/modeling/feature_engineering.md
Comment (sumane81) [inline python/spec/src/robyn/modeling/feature_engineering.md:1]: can you change spec folder name to docs? Also, remove src folder from the path

**Review by sumane81 — CHANGES_REQUESTED** [review]

**Comment by alxlyj** [comment]
see https://github.com/facebookexperimental/Robyn/pull/1110

**Pull request closed without merging**